src/main/resources/handlebars/JavaSpring/libraries/spring-boot/pom.mustache

```diff
@@ -63,24 +63,35 @@
         </dependency>
         {{/useOas2}}
         {{^useOas2}}
+        {{#springBootV2}}
         <dependency>
             <groupId>io.springfox</groupId>
             <artifactId>springfox-oas</artifactId>
             <version>${springfox-version}</version>
         </dependency>
+        <dependency>
+            <groupId>io.springfox</groupId>
+            <artifactId>springfox-swagger-ui</artifactId>
+            <version>${springfox-version}</version>
+        </dependency>
+        {{/springBootV2}}
         {{/useOas2}}
+        {{#useOas2}}
         <dependency>
             <groupId>io.springfox</groupId>
             <artifactId>springfox-swagger-ui</artifactId>
             <version>${springfox-version}</version>
-        </dependency>{{/useOas2}}
+        </dependency>
+        {{/useOas2}}
         {{^useOas2}}
+        {{^springBootV2}}
```

Comment:
Not related to change but there is an issue since `springdoc-openapi-ui` dependency is introduced but the version is conditional on `{{springBootV2}}`, fixing that as well.